src/js/wall/__tests__/LoginForm.test.tsx

```diff
@@ -0,0 +1,78 @@
+import { screen, waitFor } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import nock from "nock";
+import React from "react";
+import { useDispatch } from "react-redux";
+import { beforeEach, describe, expect, it, vi } from "vitest";
+import { renderWithProviders } from "../../../tests/setupTests";
+import LoginForm from "../LoginForm";
+
+vi.mock("react-redux", () => ({
+    useDispatch: vi.fn(),
+}));
+
+describe("<LoginForm />", () => {
+    let mockDispatch;
+
+    beforeEach(() => {
+        mockDispatch = vi.fn();
+        (useDispatch as jest.Mock).mockReturnValue(mockDispatch);
+
+        window.virtool = {
+            b2c: {
+                use: false,
+            },
+        };
+    });
+
+    afterEach(() => nock.cleanAll());
+
+    it("should call mutate() with correct values when submitted", async () => {
+        const username = "test_Username";
+        const password = "Password";
+
+        const scope = nock("http://localhost")
+            .post("/api/login", { username: username, password: password, remember: true })
+            .reply(200);
+
+        renderWithProviders(<LoginForm />);
+
+        const usernameField = screen.getByLabelText("Username");
+        await userEvent.type(usernameField, username);
+        expect(usernameField).toHaveValue(username);
+
+        const passwordField = screen.getByLabelText("Password");
+        await userEvent.type(passwordField, password);
+        expect(passwordField).toHaveValue(password);
+
+        expect(screen.getByLabelText("Remember Me")).toHaveAttribute("data-state", "unchecked");
+        await userEvent.click(screen.getByLabelText("Remember Me"));
+        expect(screen.getByLabelText("Remember Me")).toHaveAttribute("data-state", "checked");
+
+        await userEvent.click(screen.getByRole("button", { name: "Login" }));
+
+        scope.isDone();
```

Comment:
```suggestion
        scope.done();
```
Is done only returns a bool indicating if it is done, `scope.done()` raises an error if the mocks are not satisfied. You will (almost) always want to use `scope.done()` since it will let you know if your mocks were actually used. Less essential when your mocks are only for data fetching, but really important for when calling the API is the end goal.